packages/esm-patient-chart-app/src/visit/visits-widget/visit-context/visit-context-header.component.tsx

```diff
@@ -0,0 +1,70 @@
+import { Button, Loading } from '@carbon/react';
+import { showModal, useFeatureFlag, useVisit } from '@openmrs/esm-framework';
+import classNames from 'classnames';
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import styles from './visit-context-header.scss';
+import VisitContextInfo from './visit-context-info.component';
+import { useVisitContextStore } from './visit-context';
+import { useSystemVisitSetting } from '@openmrs/esm-patient-common-lib/src';
+
+interface VisitContextHeaderProps {
+  patientUuid: string;
+}
+
+const VisitContextHeader: React.FC<VisitContextHeaderProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const { systemVisitEnabled } = useSystemVisitSetting();
+  const isRdeEnabled = useFeatureFlag('rde');
+  const showVisitContextHeader = systemVisitEnabled && isRdeEnabled;
+
+  const { currentVisit, isLoading } = useVisit(showVisitContextHeader ? patientUuid : null);
+  const { manuallySetVisitUuid, setVisitContext } = useVisitContextStore();
+  const isActiveVisit = !Boolean(currentVisit && currentVisit.stopDatetime);
+
+  // This hook is needed because the `useVisit` hook sometimes
+  // returns a currentVisit that isn't actually the one in the visit context.
+  // TODO: move this into the useVisit hook
+  useEffect(() => {
+    if (showVisitContextHeader && !isLoading && currentVisit.uuid != manuallySetVisitUuid) {
```

Comment:
```suggestion
    if (showVisitContextHeader && !isLoading && currentVisit.uuid !== manuallySetVisitUuid) {
```

Yeah, can we be consistent with the strict equality checks?